src/site/antora/modules/ROOT/pages/manual/filters.adoc

```diff
@@ -17,1038 +17,1407 @@
-=== MarkerFilter
+==== `MarkerFilter`
 
-The MarkerFilter compares the configured Marker value against the Marker that is included in the LogEvent.
-A match occurs when the Marker name matches either the Log Event's Marker or one of its parents.
+The `MarkerFilter` matches log events marked with a specific marker or **any** of its descendants.
 
-.Marker Filter Parameters
-[cols="1m,1,4"]
+Besides the common configuration attributes, the `MarkerFilter` supports the following parameter:
+
+.`MarkerFilter`—configuration attributes
+[cols="1m,1,1,4"]
 |===
-|Parameter Name |Type |Description
+|Attribute | Type | Default value | Description
 
-|marker
-|String
-|The name of the Marker to compare.
+| marker
+| link:../javadoc/log4j-api/org/apache/logging/log4j/Marker.html[`Marker`]
+|
+| The filter only matches log events of marker with the given marker or one of its descendants.
 
-|onMatch
-|String
-|Action to take when the filter matches. Can be ACCEPT,
-DENY or NEUTRAL. The default value is NEUTRAL.
+**Required**
 
-|onMismatch
-|String
-|Action to take when the filter does not match. Can
-be ACCEPT, DENY or NEUTRAL. The default value is DENY.
 |===
 
-A sample configuration that only allows the event to be written by the appender if the Marker matches:
+xref:plugin-reference.adoc#org-apache-logging-log4j_log4j-core_org-apache-logging-log4j-core-filter-MarkerFilter[📖 Plugin reference for `MarkerFilter`]
 
-[source,xml]
-----
-<?xml version="1.0" encoding="UTF-8"?>
-<Configuration status="warn" name="MyApp">
-  <Appenders>
-    <RollingFile name="RollingFile" fileName="logs/app.log"
-                 filePattern="logs/app-%d{MM-dd-yyyy}.log.gz">
-      <MarkerFilter marker="FLOW" onMatch="ACCEPT" onMismatch="DENY"/>
-      <PatternLayout>
-        <pattern>%d %p %c{1.} [%t] %m%n</pattern>
-      </PatternLayout>
-      <TimeBasedTriggeringPolicy />
-    </RollingFile>
-  </Appenders>
-  <Loggers>
-    <Root level="error">
-      <AppenderRef ref="RollingFile"/>
-    </Root>
-  </Loggers>
-</Configuration>
-----
+[#message-filters]
+=== Message filters
 
-[#MutableThreadContextMapFilter]
-=== MutableThreadContextMapFilter
+Message filters allow filtering log events based on the
+xref:manual/messages.adoc[`Message`]
+contained in the log event.
 
-The MutableThreadContextMapFilter or MutableContextMapFilter allows filtering against data elements that are in the current context.
-By default, this is the ThreadContext Map.
-The values to compare are defined externally and can be periodically polled for changes.
+include::partial$manual/log-event.adoc[]
 
-.Mutable Context Map Filter Parameters
-[cols="1m,1,4"]
+[#RegexFilter]
+==== `RegexFilter`
+
+The `RegexFilter` matches a regular expression against either the result of
+link:../javadoc/log4j-api/org/apache/logging/log4j/message/Message.html#getFormat()[`Message.getFormat()`]
+or
+link:../javadoc/log4j-api/org/apache/logging/log4j/message/Message.html#getFormat()[`Message.getFormattedMessage()`].
```

Comment:
Shouldn't we document (also in `StringMatchFilter`) that this filter eagerly formats a message, which can be an expensive operation? Consider: `log.info("foo {}", () -> expensiveOp())`.

Reply:
Fixed in 4d0d7aa6c4c2ca8f9ad92621e50f8b513e59ec71.